admin/ci-tools/cleanup-vagrant-boxes

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+
+"""
+A tool for use by the CI system to clean up old versions of Vagrant boxes that
+get downloaded as part of CI work.  This helps keep disks of slaves from
+filling up with useless old boxes and eventually failing when there are too
+many.
+
+The newest version of the box is kept.
+
+Usage::
+
+  cleanup-vagrant-boxes <name of box>
+"""
+
+from subprocess import check_output
+
+
+class Box(object):
+    def __init__(self, name, rest):
+        self.name = name
+        self._rest = rest
+
+    def remove(self):
+        # self._rest is like "(virtualbox, 0.3.2.doc1.2008.g63e63f2)"
+        box_version = self._rest[1:-1].split()[1]
+        check_output([
+            b"vagrant", b"box", b"remove",
+            b"--box-version", box_version, self.name,
+        ])
+
+
+def list_vagrant_boxes():
+    boxes = check_output([b"vagrant", b"box", b"list"])
+    return (Box(name, rest) for (name, rest) in boxes.splitlines())
+
+
+def main(box_name):
+    boxes = list(
+        box for box in list_vagrant_boxes() if box.name == box_name
+    )
+    boxes.sort(lambda box: box.version.split(b"."))
```

Comment:
> Won't this mean we're alphabetically sorting versions?

Yea-ish.  I was being lazy and thought I convinced myself it would be good enough.  Certainly it's not generally correct and I guess that means it will probably bite us some day.  I guess maybe there's a function in distutils somewhere that knows how to parse these strings into something ordered?  I'll go look.


> Edit: Also, I don't see a .version property on Box. Am I missing something?

Huh ... That seems like a pretty drastic oversight.  I think the explanation must be that my "testing" of this actually never progressed beyond the single-version case.  And I guess Python doesn't bother ever calling the key function if you try to sort a list of length one.  My bad. :/